### PR TITLE
feat(DTFS2-6182): replace update party graphql endpoint with REST endpoint 

### PR DESCRIPTION
**Pull request by francescastocco**
## Introduction
We need to standardise the TFM API by moving away from having both gql and REST endpoints. We will move the gql endpoints to REST.

## Resolution
Replace the update party graphql endpoint with a REST endpoint as part of the gql -> rest migration in TFM API.

### CODE REVIEW

File: trade-finance-manager-api/src/v1/controllers/party.controller.js [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+    tfm: {
+      parties: req.body,
+    },
+  };
+
+  try {
+    const updatedDeal = await api.updateDeal(dealId, partyUpdate);
+
+    if (updatedDeal.dealSnapshot) {
+      if (await canDealBeSubmittedToACBS(updatedDeal.dealSnapshot.submissionType)) {
```
Comment (ChristianMcc) [inline trade-finance-manager-api/src/v1/controllers/party.controller.js:16]: I'm guessing was just copy pasted but I think this isn't an async function and doesn't need to be awaited

File: trade-finance-manager-api/src/v1/controllers/tfm.controller.js [hunk 2]
```diff
@@ -1,6 +1,7 @@
 const activity = require('../helpers/activity');
 const api = require('../api');
 
+// This is never used - should I delete it?
```
Comment (ChristianMcc) [inline trade-finance-manager-api/src/v1/controllers/tfm.controller.js:4]: Ah yes the function from deal.controller.js is the only one used right? Happy for this to be deleted

**Review by ChristianMcc — CHANGES_REQUESTED** [review]
Minor async change comment then can merge :)

**Review by AlexBramhill — COMMENTED** [review]
Review non-mappings files

**Pull request merged**